# Info644 Final Project - Videogame Sales

## Introduction

In [1]:
%%HTML

<div style="width:100%;height:0;padding-bottom:75%;position:relative;"><iframe src="https://giphy.com/embed/aX0RqLt2ARSW4" width="100%" height="100%" style="position:absolute" frameBorder="0" class="giphy-embed" allowFullScreen></iframe></div><p><a href="https://giphy.com/gifs/video-games-nintendo-mario-aX0RqLt2ARSW4">via GIPHY</a></p>

In 1972, the electrnoic game 'Pong' was released by the American game manufacturer Atari, Inc and took the world by storm. Since then, the video game industry has grown at an exponential rate as the video games global market is forecast to worth $256.97 billion by 2025 (Source: Mordor Intelligence). 

Video game addiction has also recently been included in the American Psychiatric Association's (APA) Diagnostic and Statistical Manual of Mental Disorders (DSM-5). Though not formally recognized as a mental disorder, APA warned viewers of the dangers of internet gaming as ""gamers" play compulsively, to the exclusion of other interests, and their persistent and recurrent online activity results in clinically significant impairment or distress." The obsession with gaming has only increased as the pandemic has left people cooped indoors.  

For the purpose of this project and stemming from my own personal interest in video games, I wanted to identify what kind of trends and patterns we can drawn from different regions of the world, and see if the data we are seeing matches the forecasted worth in global sales. The dataset I have used is a video games sales dataset from a web scrape of www.vgchartz.com. This dataset contains a list of video games with sales greater than 10,000 copies. Each record includes a breakdown of the video game’s name, rank, platform, year, genre, publisher, and sales (in North America, Europe, Japan, and worldwide). There are 11 fields and over 16,598 records. 

I used python to create several scripts that can extract different sets of data to see if there are compelling patterns and trends I can draw on from over 16,000 records. Some analyses include looking at global sales and pulling out specific video games and their sale records. I used Tableau to create all my visualizations and embedded them into this jupyter notebook.

The first python script was used to count the number of videogames sold and look at the top 10 most sold video games. The second python script was used to extract data and looking at the max games sold by region, platforms, and genres. 

The dataset can be found here: https://github.com/GregorUT/vgchartzScrape.

Info on video games addiction APA research can be found here: https://www.psychiatry.org/patients-families/internet-gaming

The full Tableau story can be seen here: https://public.tableau.com/shared/2SFYY87J5?:display_count=y&:origin=viz_share_link

For exploring and analysing data, pandas library was used.
For visualization, tableau public and wordcloud was used. 

## Importing libraries and reading CSV files with pandas

In [2]:
# let's begin by importing the basic libraries

import numpy as nb
import pandas as pd #data processing csv file
import csv

# open, analyze, and read the .csv file and turn it into a normal Pandas Data Frame

vgsales = pd.read_csv('/Users/janetliu/Desktop/videogames_sales/vgsales.csv')

## Checking and cleaning data values

In [3]:
# we can then check the first 10 rows to see what our data looks like

vgsales.head(10)

Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

In [131]:
# I am going to do a quick search to see how many rows are included in the dataset

print('Total reports: ' + str(len(vgsales)))

Total reports: 16598


In [132]:
#let's check what data types we're working with. As we can see, we have a mix of integers, objects, and floats.  

vgsales.dtypes

Rank              int64
Name             object
Platform         object
Year            float64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

In [133]:
# I'm going to change some objects into categories. 

vgsales.dtypes

for col in [
    'Platform','Genre', 'Publisher']: vgsales[col] = vgsales[col].astype('category')

In [134]:
# then re-print the datatypes to make sure the changes were made

vgsales.dtypes

Rank               int64
Name              object
Platform        category
Year             float64
Genre           category
Publisher       category
NA_Sales         float64
EU_Sales         float64
JP_Sales         float64
Other_Sales      float64
Global_Sales     float64
dtype: object

In [4]:
#let's check if our dataset include null values. We can see we have 271 in the "Year" column, and 58 in the "Publisher" column.
vgsales.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [136]:
#we can use dropna() function to remove missing values since we have a goood number of records, and set how='any' to remove a row or column with any NA values present.
vgsales.dropna(how='any',inplace=True)
vgsales

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16291 rows x 11 columns]

In [5]:
#now we can see our new dataset without nulls and how many rows were dropped. The new total row is now 16291. 
vgsales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
Rank            16598 non-null int64
Name            16598 non-null object
Platform        16598 non-null object
Year            16327 non-null float64
Genre           16598 non-null object
Publisher       16540 non-null object
NA_Sales        16598 non-null float64
EU_Sales        16598 non-null float64
JP_Sales        16598 non-null float64
Other_Sales     16598 non-null float64
Global_Sales    16598 non-null float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


## Writing new CSV file with pandas

In [27]:
#With my new dataset, lets save the modified file into our local drive so we can prepare our python scripts and embed it with charts created on Tableau Public Story.
vgsales.to_csv('vgsales_modified.csv')


## Python Scripts with embedded Tableau Charts

## Top 10s

Top 10 Global Video Games

In [6]:
 #the first script is to find out what the toop 10 most sold games are
vgsales.groupby('Name')['Global_Sales'].max().sort_values(ascending=False).head(10)

Name
Wii Sports                   82.74
Super Mario Bros.            40.24
Mario Kart Wii               35.82
Wii Sports Resort            33.00
Pokemon Red/Pokemon Blue     31.37
Tetris                       30.26
New Super Mario Bros.        30.01
Wii Play                     29.02
New Super Mario Bros. Wii    28.62
Duck Hunt                    28.31
Name: Global_Sales, dtype: float64

In [13]:
%%HTML

<div class='tableauPlaceholder' id='viz1620288558321' style='position: relative'><noscript><a href='#'><img alt='Top 10 Global Video Games ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top10GlobalGames&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;Top10GlobalGames' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top10GlobalGames&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620288558321');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

From the chart, we can see Wii sports gained 82.74 million in global sales, followed by Super Mario Bros (40.24 million), Mario Kart Wii (35.82 million), and Wii Sports resort (33.0 million).

In [14]:
%%HTML

<div class='tableauPlaceholder' id='viz1620288464725' style='position: relative'><noscript><a href='#'><img alt='Top 10 Global Video Games ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top10GlobalGamesbyRegion&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;Top10GlobalGamesbyRegion' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top10GlobalGamesbyRegion&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620288464725');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

If we look at the top 10 global games by region, we can see most of the sale of Wii sports (41.49 million) and Super Mario Bros (29.08 million) were from North America alone.

In [15]:
%%HTML

<div class='tableauPlaceholder' id='viz1620288790613' style='position: relative'><noscript><a href='#'><img alt='Top 10 Global Games by Genre, Publisher, and Platform ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top10GlobalGamesbyGenrePublisherandPlatform&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;Top10GlobalGamesbyGenrePublisherandPlatform' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top10GlobalGamesbyGenrePublisherandPlatform&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620288790613');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

The top 10 global games come from 3 publishers and belong in 7 genres. "Activism" has produced a popular game for each of it's platform with the well-known shooter game, Call of Duty. "Nintendo" has been producing a variety of genres including its two most popular series on New Super Mario and Wii Sports. The last publisher is "Take-Two Interactive" with its action game Grand Theft Auto. 

## Most Popular

Most Popular Genres

In [143]:
# As the dataset contains over 16,000 records, I wanted to write Python scripts that can pull out certain data on value counts. The first is to see what the video game genre types are, and how many top ranked video games are listed under each.
vgsales["Genre"].value_counts().sort_values(ascending=False).head(10)

Action          3251
Sports          2304
Misc            1686
Role-Playing    1470
Shooter         1282
Adventure       1274
Racing          1225
Platform         875
Simulation       848
Fighting         836
Name: Genre, dtype: int64

In [20]:
%%HTML

<div class='tableauPlaceholder' id='viz1620288949767' style='position: relative'><noscript><a href='#'><img alt='Most Popular Genres ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;MostPopularGenres_1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;MostPopularGenres_1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;MostPopularGenres_1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620288949767');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

As we can see from the chart about, 3,251 of the most popular video games belong in the action genre and 2,304 belong in the sports genre. This data indicates that gamers are interested in games that require movement and speed. 

Most Popular Publishers & Platforms

In [145]:
#number of publishers sorted by count
vgsales["Publisher"].value_counts().sort_values(ascending=False).head(10)

Electronic Arts                 1339
Activision                       966
Namco Bandai Games               928
Ubisoft                          918
Konami Digital Entertainment     823
THQ                              712
Nintendo                         696
Sony Computer Entertainment      682
Sega                             632
Take-Two Interactive             412
Name: Publisher, dtype: int64

In [146]:
#number of platforms sorted by count
vgsales["Platform"].value_counts().sort_values(ascending=False).head(10)

DS      2131
PS2     2127
PS3     1304
Wii     1290
X360    1234
PSP     1197
PS      1189
PC       938
XB       803
GBA      786
Name: Platform, dtype: int64

In [26]:
%%HTML

<div class='tableauPlaceholder' id='viz1620289203693' style='position: relative'><noscript><a href='#'><img alt='Most Popular Publishers and Platforms ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;MostPopularPublishersandPlatforms&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;MostPopularPublishersandPlatforms' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;MostPopularPublishersandPlatforms&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620289203693');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Here is a graph that shows us the vast number of popular games that have been released by various publishers. My above graph only shows 10 different publishers, but it already contains a massive scale. We can see the publihsers "Namco" and "Sega" are some of the top publishers that have continiously released a variety of games for different platforms. 

## Sales over Time

Total Sales by Region

In [18]:
#from the total sums by region, we can see most of the sales for the most popular games are from North America. It would be interesting if we could find other data that shows video game sales for other Asian countries, as the APA research conducted on online game addiction was primarily on Asian countries.
vgsales.JP_Sales.sum()

1291.0200000000002

In [149]:
vgsales.NA_Sales.sum()

4327.650000000001

In [150]:
vgsales.EU_Sales.sum()

2406.69

Yearly Sales by Region

In [152]:
vgsales.groupby(['Year'])['JP_Sales', 'EU_Sales', 'NA_Sales', 'Other_Sales'].sum()

JP_Sales  EU_Sales  NA_Sales  Other_Sales
Year                                             
1980.0      0.00      0.67     10.59         0.12
1981.0      0.00      1.96     33.40         0.32
1982.0      0.00      1.65     26.92         0.31
1983.0      8.10      0.80      7.76         0.14
1984.0     14.27      2.10     33.28         0.70
1985.0     14.56      4.74     33.73         0.92
1986.0     19.81      2.84     12.50         1.93
1987.0     11.63      1.41      8.46         0.20
1988.0     15.76      6.59     23.87         0.99
1989.0     18.36      8.44     45.15         1.50
1990.0     14.88      7.63     25.46         1.40
1991.0     14.78      3.95     12.76         0.74
1992.0     28.91     11.71     33.87         1.65
1993.0     25.33      4.65     15.12         0.89
1994.0     33.99     14.88     28.15         2.20
1995.0     45.75     14.90     24.82         2.64
1996.0     57.44     47.26     86.76         7.69
1997.0     48.87     48.32     94.75         9.13
1998.0     50.04     66.90    128.36        11.03
1999.0     52.34     62.67    126.06        10.05
2000.0     42.77     52.75     94.49        11.62
2001.0     39.86     94.89    173.98        22.76
2002.0     41.76    109.74    216.19        27.28
2003.0     34.20    103.81    193.59        26.01
2004.0     41.65    105.92    218.80        47.24
2005.0     54.28    121.56    241.58        40.55
2006.0     73.73    129.24    263.12        54.43
2007.0     60.29    160.18    311.18        77.58
2008.0     60.26    184.40    351.44        82.39
2009.0     61.89    191.59    338.85        74.77
2010.0     59.49    176.57    304.24        59.90
2011.0     53.04    167.31    241.00        54.39
2012.0     51.74    118.76    154.93        37.82
2013.0     47.59    125.80    154.77        39.82
2014.0     39.46    125.63    131.97        40.02
2015.0     33.72     97.71    102.82        30.01
2016.0     13.67     26.76     22.66         7.75
2017.0      0.05      0.00      0.00         0.00
2020.0      0.00      0.00      0.27         0.02

In [19]:
%%HTML

<div class='tableauPlaceholder' id='viz1620289919487' style='position: relative'><noscript><a href='#'><img alt='Yearly Sales by Region ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;YearlySalesbyRegion&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;YearlySalesbyRegion' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;YearlySalesbyRegion&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620289919487');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

We can see the years of 2006 to 2009 brought the most video game sales for all regions. 2008 seemed to be a good year, as North America had 351 million in sales. Since then, there has been a steep decline than exponentially growing as the market has forecasted. This rise in sales may have been contributed by Wii, as the console was released on November 19, 2006.

Top 50 Game Sales over Time

In [154]:
vgsales.groupby('Year')['Global_Sales'].max().sort_values(ascending=False)

Year
2006.0    82.74
1985.0    40.24
2008.0    35.82
2009.0    33.00
1996.0    31.37
1989.0    30.26
1984.0    28.31
2005.0    24.76
1999.0    23.10
2007.0    22.72
2010.0    21.82
2013.0    21.40
2004.0    20.81
1990.0    20.61
1988.0    17.28
2002.0    16.15
2001.0    14.98
2011.0    14.76
1998.0    14.64
2015.0    14.24
2012.0    14.03
2014.0    11.98
1992.0    11.18
1997.0    10.95
1993.0    10.55
1994.0     9.30
1982.0     7.81
2003.0     7.20
1986.0     6.51
2000.0     6.39
1995.0     5.15
2016.0     4.77
1991.0     4.61
1981.0     4.50
1987.0     4.38
1980.0     4.31
1983.0     3.20
2020.0     0.29
2017.0     0.03
Name: Global_Sales, dtype: float64

In [21]:
%%HTML

<div class='tableauPlaceholder' id='viz1620290100580' style='position: relative'><noscript><a href='#'><img alt='Top 50 Game Sales over Time ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top50GameSalesoverTime&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;Top50GameSalesoverTime' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;Top50GameSalesoverTime&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620290100580');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1000px';vizElement.style.height='627px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

If we look at the top 50 game sales for the top 10 years, we can see 2006 has been a good year that produced a variety of game genres for video gamers. These include the Wii platform games that became a big hit (again, the console was released in 2006). Action games have been doing well over time. It was a big hit in the early years of 2002 with the release of grand theft auto, followed by shooter games in the more recent years such as call of duty. It would be good for the gaming industry to produce more sport games, since the sport genre that has been doing suprisingly well. 

## Most Sold Games/Genres by Region

Most sold games by Region

In [156]:
 ##here is breakdown of the most sold games by region, starting with North America
vgsales.groupby('Name')['NA_Sales'].max().sort_values(ascending=False).head(10)

Name
Wii Sports                   41.49
Super Mario Bros.            29.08
Duck Hunt                    26.93
Tetris                       23.20
Mario Kart Wii               15.85
Wii Sports Resort            15.75
Kinect Adventures!           14.97
New Super Mario Bros. Wii    14.59
Wii Play                     14.03
Super Mario World            12.78
Name: NA_Sales, dtype: float64

In [157]:
 ##most sold games in Europe
vgsales.groupby('Name')['EU_Sales'].max().sort_values(ascending=False).head(10)

Name
Wii Sports                                      29.02
Mario Kart Wii                                  12.88
Wii Sports Resort                               11.01
Nintendogs                                      11.00
Grand Theft Auto V                               9.27
Brain Age: Train Your Brain in Minutes a Day     9.26
New Super Mario Bros.                            9.23
Wii Play                                         9.20
Pokemon Red/Pokemon Blue                         8.89
Wii Fit Plus                                     8.59
Name: EU_Sales, dtype: float64

In [158]:
 ##and most sold games in Japan
vgsales.groupby('Name')['JP_Sales'].max().sort_values(ascending=False).head(10)

Name
Pokemon Red/Pokemon Blue                       10.22
Pokemon Gold/Pokemon Silver                     7.20
Super Mario Bros.                               6.81
New Super Mario Bros.                           6.50
Pokemon Diamond/Pokemon Pearl                   6.04
Pokemon Black/Pokemon White                     5.65
Pokemon Ruby/Pokemon Sapphire                   5.38
Animal Crossing: Wild World                     5.33
Brain Age 2: More Training in Minutes a Day     5.32
Monster Hunter Freedom 3                        4.87
Name: JP_Sales, dtype: float64

Most Sold Games by Genre

In [159]:
 ##and here is a breakdown of the most sold game genres by region, starting with North America
vgsales.groupby('Genre')['NA_Sales'].max().sort_values(ascending=False).head(10)

Genre
Sports          41.49
Platform        29.08
Shooter         26.93
Puzzle          23.20
Racing          15.85
Misc            14.97
Role-Playing    11.27
Action           9.63
Simulation       9.07
Fighting         6.75
Name: NA_Sales, dtype: float64

In [160]:
 ##wmost sold game genres by Europe
vgsales.groupby('Genre')['EU_Sales'].max().sort_values(ascending=False).head(10)

Genre
Sports          29.02
Racing          12.88
Simulation      11.00
Action           9.27
Misc             9.26
Platform         9.23
Role-Playing     8.89
Shooter          5.88
Puzzle           5.36
Adventure        2.79
Name: EU_Sales, dtype: float64

In [161]:
 ##and most sold game genres by Japan
vgsales.groupby('Genre')['JP_Sales'].max().sort_values(ascending=False).head(10)

Genre
Role-Playing    10.22
Platform         6.81
Simulation       5.33
Puzzle           5.32
Misc             4.16
Racing           4.13
Action           3.96
Sports           3.77
Fighting         2.87
Adventure        2.69
Name: JP_Sales, dtype: float64

In [22]:
%%HTML

<div class='tableauPlaceholder' id='viz1619942460510' style='position: relative'><noscript><a href='#'><img alt='Most Sold Games&#47;Genres by Region ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;MostSoldGamesGenresbyRegion&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;MostSoldGamesGenresbyRegion' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;MostSoldGamesGenresbyRegion&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1619942460510');                    var vizElement = divElement.getElementsByTagName('object')[0];                    if ( divElement.offsetWidth > 800 ) { vizElement.style.minWidth='420px';vizElement.style.maxWidth='1000px';vizElement.style.width='100%';vizElement.style.minHeight='587px';vizElement.style.maxHeight='1227px';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.minWidth='420px';vizElement.style.maxWidth='1000px';vizElement.style.width='100%';vizElement.style.minHeight='587px';vizElement.style.maxHeight='1227px';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';} else { vizElement.style.width='100%';vizElement.style.height='927px';}                     var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

We can combine the most sold games/genres by region and can see these games vary greatly. North America and Europe tend to have a variety of popular games from the action, racing, shooting, and sports category. While North America has a preference for WII Sports (41.49 million in sales), we can see Europe (huge football fans) enjoy FIFA. Surprisngly, Japan gamers have a completely different preference with little interest in action and shooting games, but prefer the role-playing genre such as the pokemon series. From this data, it would seem the video game industry will do well releasing shooting or sport games that would gain big sales in Europe and North America as well as role-playing games for Japan.

## Top 100 Games Wordcloud

In [113]:
%%HTML

<a href="https://www.wordclouds.com/?yygpKSi20tfP1MvMTS8t0kvOz9WPqsoyNaxy0SvISwcA" target="_blank"><img src="https://i.imgur.com/Zzj51zD.png" border="0" alt="Made with WordClouds.com"/></a>

To wrap it up, here is a word cloud that pulls text from the top 100 most popular games. We can see the greatest count in text is wii, mario, and shooter.

## Tableau Story

In [25]:
##view the full tableau story below

In [24]:
%%HTML

<div class='tableauPlaceholder' id='viz1620290879946' style='position: relative'><noscript><a href='#'><img alt='Video Game Sales ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;VideoGameSales&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='publisherplatform&#47;VideoGameSales' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;pu&#47;publisherplatform&#47;VideoGameSales&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1620290879946');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='1016px';vizElement.style.height='991px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>